packages/core/src/components/TreeView/TreeView.js

```diff
@@ -5,33 +5,68 @@ import { noop } from '../../common/helpers';
 
 import TreeViewNode from './TreeViewNode';
 
-const TreeView = ({
-  nodes,
-  selectNode,
-  highlightOnSelect,
-  highlightOnHover
-}) => {
-  const classes = classNames('list-group', {
-    'treeview-pf-select': highlightOnSelect,
-    'treeview-pf-hover': highlightOnHover
-  });
-
-  return (
-    <div className="treeview">
-      <ul className={classes}>
-        {nodes &&
-          nodes.map((node, index) => (
-            <TreeViewNode
-              node={node}
-              key={index}
-              level={1}
-              selectNode={selectNode}
-            />
-          ))}
-      </ul>
-    </div>
-  );
-};
+class TreeView extends React.Component {
+  state = {
+    focusedNodeId: ''
+  };
+
+  onFocus = node => {
+    this.setState(() => ({ focusedNodeId: node.id }));
+  };
+
+  onKeyDown = event => {
+    const nodes = this.getVisibleNodes([...this.treeRef.current.children]);
+    const currentNodePosition = nodes.findIndex(
+      element => element.id === this.state.focusedNodeId
+    );
+    if (event.key === 'ArrowDown' && currentNodePosition !== nodes.length - 1) {
+      nodes[currentNodePosition + 1].focus();
+    } else if (event.key === 'ArrowUp' && currentNodePosition !== 0) {
+      nodes[currentNodePosition - 1].focus();
+    }
+  };
+
+  getVisibleNodes = nodes =>
+    nodes.filter(node => !node.className.match(/node-hidden/));
+
+  treeRef = React.createRef();
+
+  render() {
+    const {
+      nodes,
+      selectNode,
+      highlightOnHover,
+      highlightOnSelect
+    } = this.props;
+    const classes = classNames('list-group', {
+      'treeview-pf-select': highlightOnSelect,
+      'treeview-pf-hover': highlightOnHover
+    });
+
+    return (
+      <div className="treeview">
+        <ul
```

Comment:
Tree's have `aria-multiselectable="false"` by default, but it may be a good idea to add this metadata explicitly, in case they enhance the experience for some readers. Same for `aria-orientation="vertical"`.

Again, we haven't really covered tree's yet from the accessibility perspective, so definitely share your findings and call out any better/worse effects you notice from adding these attributes.

https://www.w3.org/TR/wai-aria-practices/#tree_roles_states_props

Reply:
I can add these `aria` attributes, but I'm worried that this will imply to PF-React consumers that there are options for multiselect and orientation